pkg/hubble/monitor/filter.go

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package monitor
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	observerTypes "github.com/cilium/cilium/pkg/hubble/observer/types"
+	"github.com/cilium/cilium/pkg/hubble/parser/errors"
+	monitorAPI "github.com/cilium/cilium/pkg/monitor/api"
+)
+
+const (
+	debug               = "debug"
+	traceSock           = "trace-sock"
+	dropNotify          = "drop-notify"
+	traceNotify         = "trace-notify"
+	policyVerdictNotify = "policy-verdict-notify"
+	debugCapture        = "debug-capture"
+	accessLog           = "access-log"
+	agent               = "agent"
+	lost                = "lost"
```

Comment:
Rather than re-defining different strings for this, could we refactor or reuse the ones in `pkg/option/runtime_options.go`? Those are already user-facing configurations for the local API, so using the same strings for agent-wide configuration could reduce confusion.

Reply:
(Changing them would also mean needing to update the `daemon/cmd/daemon_main.go` usage message)

Reply:
Not everything in this PR exists in `runtime_options.go`. There's also https://github.com/cilium/cilium/blob/master/pkg/monitor/api/types.go#L59-L69 which this PR already uses that could be used too. It's missing some options, but we could extend the names.

Reply:
For the options that end up configuring the same underlying event, we really should expose a consistent name for it in the UX. It's fine to pick something for the other ones that don't exist in runtime options, if we end up extending that in future then we can make that one follow the same definition as this.

What I'd like to avoid is that someone switches between these two different configuration mechanisms and has to refer to them by different names because we just arbitrarily decided the same thing should be named differently in different configuration mechanisms.

Reply:
(Other option is to accept both options in both interfaces, but I think it's simpler to just pick the ones that exist and make this new code use the same user-facing option names for the ones that collide)

Reply:
I guess what you're saying above is there's already multiple names for these options, hmm. The ones from https://github.com/cilium/cilium/blob/master/pkg/monitor/api/types.go#L59-L69 seem like the ones that are closest to a cluster-wide API since that shows up in Hubble messages that are sent off-box. I'd favor using those ones over runtime options ones since those are only exposed in the local API and it's pretty hidden / less likely that users are encountering those names.

Reply:
TL;DR it's suboptimal that we have two names for these options, maybe let's try to avoid three names for the same options?

Reply:
Updated to use the types from https://github.com/cilium/cilium/blob/master/pkg/monitor/api/types.go#L59-L69


Also, switching to an allowlist instead of a denylist, allowed me to remove the filter for `LostEvent`